### ISSUE TITLE
login requested for viewing pdf (inline)

### ISSUE THREAD

**Issue by jknockaert** [owncloud/core#23489]
### Steps to reproduce
1. Click on a pdf file for inline viewing
2. After pdf.js opens (and before the file is displayed) a username and password for "sabre/dav" is asked.

### Expected behaviour
No username/password should be asked; the file should be displayed inline in pdf.js. Note that downloading the file works fine; this only happens when viewing the file inline.

### Actual behaviour
Username/password are asked for.

### Server configuration
**Operating system**: Ubuntu 14.04.04 LTS
**Web server:** Nginx
**Database:** Postgres
**PHP version:** 5.5.9
**ownCloud version:** 9.0.0
**Updated from an older ownCloud or fresh install:** Updated
**Where did you install ownCloud from:** Packages
**Signing status (ownCloud 9.0 and above):** Backported #23218; no other changes
**List of activated apps:** activity: 2.2.1; calendar: 1.0; comments: 0.2; contacts: 1.0.0.0; dav: 0.1.5; encryption: 1.2.0; federatedfilesharing: 0.1.0; federation: 0.0.4; files: 1.4.4; files_external: 0.5.2; files_pdfviewer: 0.8; files_sharing: 0.9.1; files_texteditor: 2.1; files_trashbin: 0.8.0; files_versions: 1.2.0; files_videoplayer: 0.9.8; firstrunwizard: 1.1; gallery: 14.5.0; notifications: 0.2.3; provisioning_api: 0.4.1; systemtags: 0.2; templateeditor: 0.1; updatenotification: 0.1.0
**The content of config/config.php:**
```
I{
    "system": {
        "instanceid": "oc77738f5734",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED***",
        "dbtype": "pgsql",
        "version": "9.0.0.19",
        "appstoreenabled": true,
        "apps_paths": [
            {
                "path": "***REMOVED***",
                "url": "\/apps",
                "writable": true
            }
        ],
        "dbname": "***REMOVED***",
        "dbhost": "***REMOVED***",
        "dbtableprefix": "***REMOVED***",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "enable_avatars": false,
        "installed": true,
        "forcessl": true,
        "loglevel": "2",
        "log_authfailip": true,
        "knowledgebaseenabled": true,
        "logtimezone": "***REMOVED***",
        "maintenance": false,
        "logfile": "***REMOVED***",
        "theme": "",
        "trusted_domains": [
            "***REMOVED***"
        ],
        "share_folder": "\/Shared",
        "enable_previews": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "mail_smtpmode": "php",
        "mail_domain": "***REMOVED***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "asset-pipeline.enabled": false,
        "trashbin_retention_obligation": "auto",
        "updatechecker": false
    }
}
```
**Are you using external storage, if yes which one:** webdav (but the behaviour reported happens for local storage)
**Are you using encryption:** yes
**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 45.0.1
**Operating system:** OS X

### Logs
Nothing

**Comment by davodego** [comment]
I have the same behavior. Click on PDF for inline viewing. pdf.js opens and webserver asks for username and password from sabre/dav.

Ubuntu 14.04 LTS
Apache Webserver
PHP 5.5.0
Owncloud 9
Mysql Database 
No encryption

Firefox 45

**Comment by jknockaert** [comment]
A further observation is that the authentication only pops up when logged in as an owncloud user. Inline viewing works fine when opening a (shared) pdf document using an url.
I have no clue as to what triggers this behaviour.

**Comment by Xenopathic** [comment]
Unable to reproduce on master with the 9.0.0 files_pdfviewer installed, on Chrome.

**Comment by jknockaert** [comment]
Tested again with Safari, and indeed this behaviour seems to be specific to Firefox (45).

**Comment by davodego** [comment]
Tested with Chromium 50 Portable 64bit, and it works as expected, pdf was shown without promt for password. It seems a firefox 45 problem.

**Comment by davodego** [comment]
Tested with InternetExplorer 11, also no problem showing PDF.

**Comment by davodego** [comment]
After I tested the issue with Chromium and InternetExplorer and all goes as expected, I go back to firefox to see if the developer terminal shows an error. 
But now the issue is gone. Firefox shows no login promt from sabre/dav.
@jknockaert please test with your firefox too.

**Comment by jknockaert** [comment]
@davodego No, issue is still there with FF45.

**Comment by davodego** [comment]
@jknockaert with my firefox 45 on windows the issue is gone. I don't have changed anything. Just logout, tested with Chromium and IE11 and the go back to firefox and login again. No restart of firefox.

**Comment by jknockaert** [comment]
@davodego I did the same on mac os x (tested with Safari). But the issue is still there in FF45.

**Comment by davodego** [comment]
Nope its back again:, logout -> close firefox -> start firefox -> login to owncloud and the issue is there again :(
I dont know what happend.

**Comment by Xenopathic** [comment]
Would it be possible for you guys to test with older versions of Firefox? Also, if you could look at the Network tab in the Firefox dev tools, you should be able to see the exact HTTP messages that are sent to the server

**Comment by jknockaert** [comment]
@Xenopathic Not sure what I should see in the dev tools. Just checking superficially I see the pdf is requested (GET) and met with a 401:
```
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\NotAuthenticated</s:exception>
  <s:message>No 'Authorization: Basic' header found. Either the client didn't send one, or the server is mis-configured</s:message>
</d:error>
```
As far as I can see no cookies are sent with the http GET request, that looks a bit suspicious.
No possibility to test with older FF versions right now.

**Comment by Xenopathic** [comment]
That's definitely the problem then, there should have been cookies sent with the GET request, to establish the session. Is the request going to the same domain as you use to access ownCloud? I'm clutching at straws, but short of a bug in Firefox that's the only thing it could be.

**Comment by jknockaert** [comment]
There's nothing wrong with the domain. When I (successfully) download the file, the call is to:
```
https://www.domain.com/remote.php/webdav/Folder/file.pdf?downloadStartSecret=***REMOVED***
```
and two cookies are sent (oc***REMOVED*** and oc_sessionPassphrase). When I view it inline, the call is to:
```
https://www.domain.com/remote.php/webdav/Folder/file.pdf
```
and no cookies are sent.

**Comment by Xenopathic** [comment]
In that case, the only possible outcome is a bug in Firefox. It should be sending the cookies, but isn't. Can you report this upstream to them?

**Comment by jknockaert** [comment]
Fixed in FF 45.0.2
